cluster/mesos/task.go

```diff
@@ -0,0 +1,162 @@
+		Docker: &mesosproto.ContainerInfo_DockerInfo{
+			Image: &t.config.Image,
+		},
+	}
+
+	switch t.config.HostConfig.NetworkMode {
+	case "none":
+		t.Container.Docker.Network = mesosproto.ContainerInfo_DockerInfo_NONE.Enum()
+	case "host":
+		t.Container.Docker.Network = mesosproto.ContainerInfo_DockerInfo_HOST.Enum()
+	case "bridge", "":
+		for containerPort, bindings := range t.config.HostConfig.PortBindings {
+			for _, binding := range bindings {
+				fmt.Println(containerPort)
+				containerInfo := strings.SplitN(containerPort, "/", 2)
+				fmt.Println(containerInfo[0], containerInfo[1])
+				containerPort, err := strconv.ParseUint(containerInfo[0], 10, 32)
+				if err != nil {
+					log.Warn(err)
+					continue
+				}
+				hostPort, err := strconv.ParseUint(binding.HostPort, 10, 32)
+				if err != nil {
+					log.Warn(err)
+					continue
+				}
+				protocol := "tcp"
+				if len(containerInfo) == 2 {
+					protocol = containerInfo[1]
+				}
+				t.Container.Docker.PortMappings = append(t.Container.Docker.PortMappings, &mesosproto.ContainerInfo_DockerInfo_PortMapping{
+					HostPort:      proto.Uint32(uint32(hostPort)),
+					ContainerPort: proto.Uint32(uint32(containerPort)),
+					Protocol:      proto.String(protocol),
+				})
+				t.Resources = append(t.Resources, mesosutil.NewRangesResource("ports", []*mesosproto.Value_Range{mesosutil.NewValueRange(hostPort, hostPort)}))
+			}
+		}
+		// TODO handle -P here
+		t.Container.Docker.Network = mesosproto.ContainerInfo_DockerInfo_BRIDGE.Enum()
+	default:
+		log.Errorf("Unsupported network mode %q", t.config.HostConfig.NetworkMode)
+		t.Container.Docker.Network = mesosproto.ContainerInfo_DockerInfo_BRIDGE.Enum()
+	}
+
+	if cpus := t.config.CpuShares; cpus > 0 {
+		t.Resources = append(t.Resources, mesosutil.NewScalarResource("cpus", float64(cpus)))
+	}
+
+	if mem := t.config.Memory; mem > 0 {
+		t.Resources = append(t.Resources, mesosutil.NewScalarResource("mem", float64(mem/1024/1024)))
+	}
+
+	if len(t.config.Cmd) > 0 && t.config.Cmd[0] != "" {
+		t.Command.Value = &t.config.Cmd[0]
+	}
+
+	if len(t.config.Cmd) > 1 {
+		t.Command.Arguments = t.config.Cmd[1:]
+	}
+
+	for key, value := range t.config.Labels {
+		t.Container.Docker.Parameters = append(t.Container.Docker.Parameters, &mesosproto.Parameter{Key: proto.String("label"), Value: proto.String(fmt.Sprintf("%s=%s", key, value))})
+	}
+
+	t.SlaveId = &mesosproto.SlaveID{Value: &slaveID}
+}
+
+func newTask(c *Cluster, config *cluster.ContainerConfig, name string) (*task, error) {
+	// save the name in labels as the mesos containerizer will override it
+	config.Labels[cluster.SwarmLabelNamespace+".mesos.name"] = name
+
+	task := task{
+		updates: make(chan *mesosproto.TaskStatus),
+
+		cluster:   c,
+		config:    config,
+		error:     make(chan error),
+		container: make(chan *cluster.Container),
+	}
+
+	ID := stringid.TruncateID(stringid.GenerateRandomID())
+	if name != "" {
+		ID = name + "." + ID
```

Comment:
It's a "common practice" in mesos, see some marathon task in the screenshot.
![screen shot 2015-05-26 at 11 26 06](https://cloud.githubusercontent.com/assets/1032519/7819970/1b5fedae-039a-11e5-98b3-70bed234bdd8.png)